File: week2/이다혜/기능개발.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.Queue;

public class Solution {

    public ArrayList<Integer> solution(int[] progresses, int[] speeds) {

        ArrayList<Integer> answer = new ArrayList<>();

        int n = progresses.length;
        int [] daysOfProgress = new int[n];
        
        for(int i = 0; i < n; i++) {
            int left = 100 - progresses[i];
            daysOfProgress[i] = (left % speeds[i] == 0 ? left / speeds[i] : left / speeds[i] + 1);
        }

        Queue<Integer> q = new LinkedList<>();
        for(int i = 0; i < n; i++) {
            if(q.isEmpty()) q.add(daysOfProgress[i]);
            else {
                if(q.peek() >= daysOfProgress[i]) q.add(daysOfProgress[i]);
                else {
                    answer.add(q.size());
                    q.clear();
                    q.add(daysOfProgress[i]);
                }
            }
        }

        if(!q.isEmpty()) answer.add(q.size());

        return answer;
    }
    
}

File: week2/윤여준/기능개발.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> p, vector<int> speeds) {
    vector<int> answer;
    
    int i = 0;
    while(i < p.size()) {
        
        for(int j = 0; j < p.size(); j++)
            p[j] += speeds[j];
        
        int temp = 0;
        while(i < p.size() && p[i] >= 100) {
            temp++; i++;
        }
            
        if(temp > 0) answer.push_back(temp);
    }
    
    return answer;
}